zerostash/src/commands/mount.rs

```diff
@@ -68,22 +71,19 @@ fn mount(
     Ok(())
 }
 
-type DirPath = PathBuf;
-type FilePath = PathBuf;
-type FileContent = Vec<u8>;
-type Metadata = Arc<Entry>;
+type ChunkData = (Vec<u8>, usize, usize);
```

Comment:
You may find it easier to communicate what's happening if you turn this into a proper `struct`, and define state transformation functions on it (`fn a(&mut self, params....)`). I think you will find that a lot of the loop code from `read()` below will fit in here nicely.